python/code/test/test_manifold_discriminant.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+from classificationmethods import ManifoldDiscrimantAnalysisParafacTucker as pt
+import numpy as np
+from numpy import random as random
+import scipy.stats as stats
+
+"""
+    Unit tests for python manifold discriminant analysis. 
+"""
+
+
+def generate_data(classes, length_of_data_set, shape=None):
+    end_padding = length_of_data_set % classes
+    number_of_copies = length_of_data_set // classes
+    output_classes = number_of_copies * list(range(classes)) + list(range(end_padding))
+    if shape is None:
+        shape = (5, 6)
+
+    observations = np.multiply.outer(output_classes, np.zeros(shape))
+
+    for i in range(classes):
```

Comment:
This loop seems to only change the first 'classes' observations away from zero. If classes is 3, only the first three observations will have random numbers added to them.